Nightly build failure: recent updates break kokkos and kokkos-kernels, cuda/12.0+gcc/11.3 w/ c++17 and c++20

**Describe the bug**

Nightly builds of kokkos and kokkos-kernels with cuda/12.0+gcc/11.3 c++20 fail to compile following recent updates in kokkos

**Error snip**
```
[ 38%] Building CXX object core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o
cd <https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/Build/kokkos-install/core/src> && <https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/bin/nvcc_wrapper> -DKOKKOS_DEPENDENCE -I<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/Build/kokkos-install/core/src> -I<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/core/src> -I<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/Build/kokkos-install> -I<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/tpls/desul/include> -I<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/tpls/mdspan/include> -O2 -g -DNDEBUG -std=c++20 -mcpu=power9 -mtune=power9 -extended-lambda -Wext-lambda-captures-this -arch=sm_70 -MD -MT core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o -c <https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/core/src/impl/Kokkos_ExecPolicy.cpp>
<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp>: In member function 'void Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType, Kokkos::RangePolicy<Traits ...>, Kokkos::Cuda>::execute()':
<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp>:310:28: error: cannot call member function 'int Kokkos::Cuda::concurrency() const' without object
  310 |       auto cc = Kokkos::Cuda::concurrency() / block_size;
      |           ~~~~~~~~~~~~~~~~~^~
<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp>: In member function 'void Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType, Kokkos::RangePolicy<Traits ...>, Kokkos::Cuda>::execute()':
<https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/ws/kokkos/core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp>:310:28: error: cannot call member function 'int Kokkos::Cuda::concurrency() const' without object
  310 |       auto cc = Kokkos::Cuda::concurrency() / block_size;
      |           ~~~~~~~~~~~~~~~~~^~
make[2]: *** [core/src/CMakeFiles/kokkoscore.dir/build.make:135: core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

This began following merge of these commits

**Changes**

Git (git https://github.com/kokkos/kokkos.git)

    Add reduction benchmark ([detail](https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/447/changes#detail))
    Improve CUDA reduction performance by using more blocks ([detail](https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/447/changes#detail))
    Enable bugprone-multi-level-implicit-pointer-conversion (#7842) ([detail](https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/447/changes#detail))
    Drop `TeamPolicyInternal<{Cuda,HIP,SYCL},...>::verify_requested_vector_length()` (#7836) ([detail](https://jenkins-son.sandia.gov/job/KokkosKernels_Weaver_Cuda_Serial_cuda_1200_gcc_1130-cpp20/447/changes#detail))

**Please include the following for a minimal reproducer**

**Reproducer notes** (weaver testbed):

```bash
bsub -Is -q rhel8 -gpu "num=1" bash

module load cmake git gcc/11.3.0 cuda/12.0.0

# Kokkos configuration
${KOKKOS_PATH}/generate_makefile.bash --with-cuda --with-serial --compiler=${KOKKOS_PATH}/bin/nvcc_wrapper --arch=Volta70,Power9 --with-cuda-options=enable_lambda --kokkos-path=${KOKKOS_PATH} --cxxstandard=20

# Kokkos Kernels configuration
${KOKKOSKERNELS_PATH}/cm_generate_makefile.bash --with-cuda --with-serial --compiler=${KOKKOS_PATH}/bin/nvcc_wrapper --arch=Volta70,Power9 --with-cuda-options=enable_lambda --kokkos-path=${KOKKOS_PATH} --kokkoskernels-path=${KOKKOSKERNELS_PATH} --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --cxxstandard=20 --kokkos-cmake-flags=-DKokkos_ENABLE_COMPLEX_ALIGN=OFF

make -j16

```

--- a/.github/workflows/snl-ci.yml
+++ b/.github/workflows/snl-ci.yml
@@ -24,7 +24,6 @@ jobs:
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           repository: kokkos/kokkos
-          ref: ${{ github.base_ref }}
           path: kokkos
 
       - name: configure_kokkos

--- a/core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp
+++ b/core/src/Cuda/Kokkos_Cuda_Parallel_Range.hpp
@@ -307,7 +307,7 @@ class ParallelReduce<CombinedFunctorReducerType, Kokkos::RangePolicy<Traits...>,
 
       // REQUIRED ( 1 , N , 1 )
       dim3 block(1, block_size, 1);
-      auto cc = Kokkos::Cuda::concurrency() / block_size;
+      auto cc = m_policy.space().concurrency() / block_size;
       dim3 grid(
           std::min(index_type(cc), index_type((nwork + block.y - 1) / block.y)),
           1, 1);

